src/main/groovy/edu/oregonstate/mist/locations/db/ArcGisDAO.groovy

```diff
@@ -60,12 +60,17 @@ class ArcGisDAO {
                 "attributes"
         )
 
-        mergeArcGisData(centroidData, genderInclusiveRRData)
+        mergeArcGisData(centroidData, genderInclusiveRRData, buildings)
     }
 
     private HashMap<String, ArcGisLocation> mergeArcGisData(
             HashMap<String, ArcGisLocation> centroidData,
-            HashMap<String, ArcGisLocation> genderInclusiveRRData) {
+            HashMap<String, ArcGisLocation> genderInclusiveRRData, List<FacilLocation> buildings) {
+
+        def buildingIDs = buildings*.bldgID
+        def centroidDataToBeRemoved =  centroidData.collect { it.value.bldID } - buildingIDs
+        logger.warn("Buildings found in centroid, but not in AIM: " + centroidDataToBeRemoved)
+        centroidData.entrySet().removeIf { ! buildingIDs.contains(it.value.bldID) }
 
         genderInclusiveRRData.each {id, value ->
             if (centroidData[id]) {
```

Comment:
We should be merging genderInclusiveRR data into buildings (facil locations), not the centroid data. If we have GIRR data for a building that's in the facil locations but not in the centroid, the data is currently being ignored. Right now, Valley sports performance center and property services are being returned by the api, but their GIRR data is not being included.